src/PipelineNet/AsyncBaseMiddlewareFlow.cs

```diff
@@ -60,5 +61,38 @@ protected void AddMiddleware(Type middlewareType)
 
             this.MiddlewareTypes.Add(middlewareType);
         }
+
+        internal void EnsureMiddlewareNotNull(MiddlewareResolverResult middlewareResolverResult, Type middlewareType)
+        {
+            if (middlewareResolverResult == null || middlewareResolverResult.Middleware == null)
+            {
+                throw new InvalidOperationException($"'{MiddlewareResolver.GetType()}' failed to resolve middleware of type '{middlewareType}'.");
+            }
+        }
+
+        internal static async Task DisposeMiddlewareAsync(MiddlewareResolverResult middlewareResolverResult)
+        {
+            if (middlewareResolverResult != null && middlewareResolverResult.Dispose)
+            {
+                var middleware = middlewareResolverResult.Middleware;
+                if (middleware != null)
+                {
+#if NETSTANDARD2_1_OR_GREATER
+                    if (middleware is IAsyncDisposable asyncDisposable)
+                    {
+                        await asyncDisposable.DisposeAsync().ConfigureAwait(false);
+                    }
+                    else
```

Comment:
Could we have a return to simplify the `if-else` with the preprocessor directives?

```suggestion
                        await asyncDisposable.DisposeAsync().ConfigureAwait(false);
                        return;
                    }
```

Reply:
How about:
```C#
        internal static async Task DisposeMiddlewareAsync(MiddlewareResolverResult middlewareResolverResult)
        {
            if (middlewareResolverResult != null && middlewareResolverResult.Dispose)
            {
                var middleware = middlewareResolverResult.Middleware;
                if (middleware != null)
                {
#if NETSTANDARD2_1_OR_GREATER
                    if (middleware is IAsyncDisposable asyncDisposable)
                    {
                        await asyncDisposable.DisposeAsync().ConfigureAwait(false);
                        return;
                    }
                    else
#endif
                    if (middleware is IDisposable disposable)
                    {
                        disposable.Dispose();
                    }

                    var completedTask = Task.FromResult(0);
                    await completedTask.ConfigureAwait(false);
                }
            }
        }
```
Would that be fine too?

Reply:
Being impatient, I just went ahead and added my version. Hope this is fine😅